engines/private/private.cpp

```diff
@@ -0,0 +1,1120 @@
+    //_pixelFormat = Graphics::PixelFormat::createFormatCLUT8();
+    _pixelFormat = Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0);
+    _transparentColor = _pixelFormat.RGBToColor(0,255,0);
+    initGraphics(_screenW, _screenH, &_pixelFormat);
+    changeCursor("default");
+
+    _origin = new Common::Point(0, 0);
+    _image = new Image::BitmapDecoder();
+    _compositeSurface = new Graphics::ManagedSurface();
+    _compositeSurface->create(_screenW, _screenH, _pixelFormat);
+    _compositeSurface->setTransparentColor(_transparentColor);
+
+    // Create debugger console. It requires GFX to be initialized
+    Console *console = new Console(this);
+    setDebugger(console);
+
+    // Additional setup.
+    debug("PrivateEngine::init");
+
+    // Simple main event loop
+    Common::Event event;
+    Common::Point mousePos;
+    _videoDecoder = nullptr; //new Video::SmackerDecoder();
+
+    int saveSlot = ConfMan.getInt("save_slot");
+    if (saveSlot >= 0) { // load the savegame
+        loadGameState(saveSlot);
+    } else {
+        _nextSetting = &kGoIntro;
+    }
+
+    while (!shouldQuit()) {
+        checkPhoneCall();
+
+        while (g_system->getEventManager()->pollEvent(event)) {
+            mousePos = g_system->getEventManager()->getMousePos();
+            // Events
+            switch (event.type) {
+            case Common::EVENT_KEYDOWN:
+                if (event.kbd.keycode == Common::KEYCODE_ESCAPE && _videoDecoder)
+                    skipVideo();
+
+                break;
+
+            case Common::EVENT_QUIT:
+            case Common::EVENT_RETURN_TO_LAUNCHER:
+                break;
+
+            case Common::EVENT_LBUTTONDOWN:
+                _numberClicks++;
+                if (selectDossierNextSuspect(mousePos))
+                    break;
+                else if (selectDossierPrevSuspect(mousePos))
+                    break;
+
+                selectPauseMovie(mousePos);
+                selectPhoneArea(mousePos);
+                selectPoliceRadioArea(mousePos);
+                selectAMRadioArea(mousePos);
+                selectLoadGame(mousePos);
+                selectSaveGame(mousePos);
+                if (!_nextSetting)
+                    selectMask(mousePos);
+                if (!_nextSetting)
+                    selectExit(mousePos);
+                break;
+
+            case Common::EVENT_MOUSEMOVE:
+                changeCursor("default");
+                if      (cursorPauseMovie(mousePos)) {}
+                else if (cursorMask(mousePos))       {}
+                else if (cursorExit(mousePos))       {}
+                //
+                break;
+
+            default:
+            {}
+
+            }
+        }
+
+        checkPoliceBust();
+
+        // Movies
+        if (_nextMovie != NULL) {
+            removeTimer();
+            _videoDecoder = new Video::SmackerDecoder();
+            playVideo(*_nextMovie);
+            _nextMovie = NULL;
+            continue;
+
```

Comment:
In general, it is best to avoid empty lines like this to conserve the vertical space. E.g. they're useful between the functional blocks which is not the case here.

Reply:
Done.